python/tests/operations/slice.py

```diff
@@ -54,26 +54,26 @@ def _check_sliced_block(self, block, sliced_block, structures_to_keep):
             # no slicing of properties has occurred
             self.assertTrue(np.all(sliced_gradient.properties == gradient.properties))
 
-            # samples have been sliced to the correct dimension
+            # samples have been updated to refer to the new samples
```

Comment:
This only checks that the max sample index isn't greater than the number of samples in the parent. Should we also check that the mapping of samples indices in the Gradient is correct too?

Reply:
I could not find a nice way to test it without re-implementing the function ... Maybe the best kind of test would be to check the gradients with finite differences? Should not be too hard to do, and we can re-use it for all operations